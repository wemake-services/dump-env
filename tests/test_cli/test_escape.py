import json


def test_quote_escape(monkeypatch, delegator):
    """Check that cli works with multiline inputs"""
    monkeypatch.setenv('MULTILINE_VALUE', json.dumps({
        'key': 'value',
        'key2': 'multi\nline\nvalue',
    }, indent=4))

    variables = delegator('dump-env -p MULTILINE_')
    assert variables == '''VALUE="{
    \\"key\\": \\"value\\",
    \\"key2\\": \\"multi\\\\nline\\\\nvalue\\"
}"
'''
