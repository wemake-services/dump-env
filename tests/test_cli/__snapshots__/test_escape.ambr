# serializer version: 1
# name: test_quote_escape
  '''
  VALUE="{
      \"key\": \"value\",
      \"key2\": \"multi\\nline\\nvalue\"
  }"
  
  '''
# ---
